maitake/src/wait/cell.rs

```diff
@@ -56,26 +91,24 @@ impl WaitCell {
 }
 
 impl WaitCell {
-    pub fn poll_wait(&self, waker: &Waker) -> Poll<WaitResult<()>> {
+    fn register_wait(&self, waker: &Waker) -> Result<(), Error> {
```

Comment:
Updated the name and signature.

The name makes more sense (to me), since you aren't really polling for the wait to be ready. You're attempting to register a new waiter to the Wait Cell.

The type also changed, because you CANNOT return `Poll::Ready(Ok(()))` here anymore, only an error or a pending.

`Ok(())` means you registered successfully, which maps to `Poll::Pending`.

Reply:
i think the name and signature changes make sense here, the `poll_wait` design was me trying to be too clever, and i think maybe it never made that much sense in most cases.

i *do* think, however, this method probably should still be `pub`. 

i can imagine wanting to register a waiter outside of a `Wait` future (for example, in a `JoinHandle`). in particular, the `Wait` future borrows the `WaitCell`, and i think a common use case might involve registering a waker with a `WaitCell` in something that doesn't have a non-`'static` lifetime --- for example, the `WaitCell` might be inside of some other shared structure that's `Arc`ed, and we might want to clone that `Arc` into the future that registers, or something... 

```suggestion
    pub fn register_wait(&self, waker: &Waker) -> Result<(), Error> {
```